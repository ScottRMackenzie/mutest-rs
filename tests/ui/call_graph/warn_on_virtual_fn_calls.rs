//@ print-targets
//@ stdout
//@ stderr

fn impl_call() {}

fn make_virtual_fn_call(f: &dyn Fn()) {
    f();
}

fn make_virtual_fn_mut_call(f: &mut dyn FnMut(usize) -> usize) {
    let _ = f(1);
}

#[test]
fn test() {
    make_virtual_fn_call(&|| impl_call());
    make_virtual_fn_mut_call(&mut |v| v + 1);

    let f: &dyn Fn() = &|| impl_call();
    f();
}
