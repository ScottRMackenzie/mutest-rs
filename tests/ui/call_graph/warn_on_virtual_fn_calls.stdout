tests -(0)-> make_virtual_fn_call at tests/ui/call_graph/warn_on_virtual_fn_calls.rs:7:1: 7:38 (#0)
  (0) test

tests -(0)-> make_virtual_fn_mut_call at tests/ui/call_graph/warn_on_virtual_fn_calls.rs:11:1: 11:63 (#0)
  (0) test

targets: 2 total; 2 safe; 0 unsafe (0 tainted)
